.github/workflows/tests.yml

```diff
@@ -277,76 +286,74 @@ jobs:
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
+          conda install -q -y -c defaults --file tests/requirements.txt
+          conda install -q -y -c defaults --file tests/requirements-macos.txt
+          conda install -q -y ${{ env.CONDA_CHANNEL_LABEL }}::conda
+          pip install -e .
 
-      - name: Setup CF MacOS Bits and Select Xcode
+      - name: Setup variables
+        # macOS needs to define $CONDA_BUILD_SYSROOT and $MACOSX_DEPLOYMENT_TARGET to know where to find the SDKs
+        # the Anaconda compilers only support up to macOS 10.15
         run: |
-          set -x -e -u
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
-          source ci/github/run_conda_forge_build_setup_osx
+          sudo xcode-select --switch /Applications/Xcode_11.7.app
+
+          echo "CONDA_BUILD_SYSROOT=$(xcrun --sdk macosx --show-sdk-path)" >> $GITHUB_ENV
+          echo "MACOSX_DEPLOYMENT_TARGET=$(xcrun --sdk macosx --show-sdk-version)" >> $GITHUB_ENV
 
-      - name: Prepare Test Environment
+      - name: Show info
         run: |
-          set -e -u
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
-          conda info
-          conda list
-          grep '^#' "${CONDA_PREFIX}/conda-meta/history"
-          conda install python=${{ matrix.python-version }} -y
-          mkdir -p ${{ github.workspace }}/miniconda/locks
-          mkdir -p ${{ github.workspace }}/miniconda/bin
-          chmod -w ${{ github.workspace }}/miniconda/locks
-          conda install -y -c conda-canary/label/dev conda
-          conda config --set always_yes yes
-          conda config --set auto_update_conda False
-          conda info
-          source ci/github/install_conda_build_test_deps
-          pip install -e .
           conda info -a
           conda list --show-channel-urls
 
-      - name: Run Serial Tests
-        if: matrix.test-type == 'serial'
-        run: |
-          set -e -u
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
-          conda install conda-verify -y
-          pytest --color=yes -v -n 0 --basetemp /tmp/cb_serial --cov conda_build --cov-report xml -m "serial" tests "${PYTEST_REPLAY_OPTIONS[@]+"${PYTEST_REPLAY_OPTIONS[@]}"}"
-
-      - name: Run Parallel Tests
-        if: matrix.test-type == 'parallel'
+      - name: Run tests
         run: |
-          set -e -u
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
-          conda remove conda-verify -y
-          echo "safety_checks: disabled" >> ~/.condarc
-          echo "local_repodata_ttl: 1800" >> ~/.condarc
-          mkdir -p ~/.conda
-          conda create -n blarg1 -yq python=2.7
-          conda create -n blarg3 -yq python=3.7
-          conda create -n blarg4 -yq python nomkl numpy pandas svn
```

Comment:
What's up with these, any idea where they came from?

Reply:
Those were added to "pre-warm" the package cache, i.e., avoid downloading those packages during the (parallel) tests, I'd guess to mostly avoid concurrency issues. At some point those had `--download-only` I think -- either I'm misremembering or those got lost somewhere one the way (with `--download-only` those env wouldn't have been created which explains why Mike/Ray(?) chose such "meaningful" names :laughing: ...)
We could leave them out and see if anything causes trouble, I'd say. If it leads to any flakiness, we could add something like this (or rather something more precise) back again.

Reply:
yeah, I was suspecting it was for caching, if we need it we should create a pytest fixture to do the caching